src/openms/source/ANALYSIS/ID/AhoCorasickDA.cpp

```diff
@@ -443,41 +844,36 @@ namespace OpenMS
 
 
     // add new empty nodes
-    for (uint32_t i = 0; i < 256; ++i)
-    {
-      bc_.push_back(BC{});
-      build_info_.push_back(BuildInformation{});
-    }
+    bc_.resize(new_size);
 
 
     // set linkage
-    for (auto i = old_size; i < new_size; i++)
+    for (auto i = old_size; i < new_size; ++i)
     {
-      build_info_[i].prev = i - 1;
-      build_info_[i].next = i + 1;
+      build_info_.emplace_back(BuildInformation_{i});
```

Comment:
`build_info_.emplace_back(i);` should do it.
Currently, you are copying the BuildInfo object. But thats not needed. Constructing it in-place (within the target vector) is better.